src/System.Net.Security/src/System/Net/SecureProtocols/SslStream.cs

```diff
@@ -0,0 +1,455 @@
+        {
+        }
+        public SslStream(Stream innerStream, bool leaveInnerStreamOpen, RemoteCertificateValidationCallback userCertificateValidationCallback,
+            LocalCertificateSelectionCallback userCertificateSelectionCallback)
+                : this(innerStream, leaveInnerStreamOpen, userCertificateValidationCallback, userCertificateSelectionCallback, EncryptionPolicy.RequireEncryption)
+        {
+        }
+        public SslStream(Stream innerStream, bool leaveInnerStreamOpen, RemoteCertificateValidationCallback userCertificateValidationCallback,
+            LocalCertificateSelectionCallback userCertificateSelectionCallback, EncryptionPolicy encryptionPolicy)
+            : base(innerStream, leaveInnerStreamOpen)
+        {
+            if (encryptionPolicy != EncryptionPolicy.RequireEncryption && encryptionPolicy != EncryptionPolicy.AllowNoEncryption && encryptionPolicy != EncryptionPolicy.NoEncryption)
+                throw new ArgumentException(SR.Format(SR.net_invalid_enum, "EncryptionPolicy"), "encryptionPolicy");
+
+            _userCertificateValidationCallback = userCertificateValidationCallback;
+            _userCertificateSelectionCallback = userCertificateSelectionCallback;
+            RemoteCertValidationCallback _userCertValidationCallbackWrapper = new RemoteCertValidationCallback(UserCertValidtaionCallbackWrapper);
+            LocalCertSelectionCallback _userCertSelectionCallbackWrapper = userCertificateSelectionCallback == null ? null : new LocalCertSelectionCallback(userCertSelectionCallbackWrapper);
+            _SslState = new SslState(innerStream, _userCertValidationCallbackWrapper, _userCertSelectionCallbackWrapper, encryptionPolicy);
+        }
+
+        private bool UserCertValidtaionCallbackWrapper(string hostName, X509Certificate2 certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
+        {
+            _remoteCertificateOrBytes = certificate == null ? null : certificate.RawData;
+            if (_userCertificateValidationCallback == null)
+            {
+                if (!_SslState.RemoteCertRequired)
+                    sslPolicyErrors &= ~SslPolicyErrors.RemoteCertificateNotAvailable;
+
+                return (sslPolicyErrors == SslPolicyErrors.None);
+            }
+            else
+                return _userCertificateValidationCallback(this, certificate, chain, sslPolicyErrors);
+        }
+
+        private X509Certificate userCertSelectionCallbackWrapper(string targetHost, X509CertificateCollection localCertificates, X509Certificate remoteCertificate, string[] acceptableIssuers)
+        {
+            return _userCertificateSelectionCallback(this, targetHost, localCertificates, remoteCertificate, acceptableIssuers);
+        }
+
+        private SslProtocols DefaultProtocols()
+        {
+            SslProtocols protocols = SslProtocols.Tls12 | SslProtocols.Tls11 | SslProtocols.Tls;
```

Comment:
Will change to const.

@SidharthNabar :  We should also make this function public and recommend that apps use it instead of hardcoding the default protocol list.